src/mon/MgrMonitor.cc

```diff
@@ -33,6 +34,102 @@ static ostream& _prefix(std::ostream *_dout, Monitor *mon,
 		<< ").mgr e" << mgrmap.get_epoch() << " ";
 }
 
+
+class MgrPGStatService : public PGMap, public PGStatService {
+  PGMap& parent;
+public:
+  MgrPGStatService() : PGMap(), PGStatService(),
+		    parent(*static_cast<PGMap*>(this)) {}
+  MgrPGStatService(const PGMap& o) : PGMap(o), PGStatService(),
+				  parent(*static_cast<PGMap*>(this)) {}
+  void reset(const PGMap& o) {
+    parent = o;
+  }
+
+  const pool_stat_t* get_pool_stat(int poolid) const {
+    auto i = parent.pg_pool_sum.find(poolid);
+    if (i != parent.pg_pool_sum.end()) {
+      return &i->second;
+    }
+    return NULL;
+  }
+  
+  const pool_stat_t& get_pg_sum() const { return parent.pg_sum; }
+  const osd_stat_t& get_osd_sum() const { return parent.osd_sum; }
+
+  const osd_stat_t *get_osd_stat(int osd) const {
+    auto i = parent.osd_stat.find(osd);
+    if (i == parent.osd_stat.end()) {
+      return NULL;
+    }
+    return &i->second;
+  }
+  const ceph::unordered_map<int32_t,osd_stat_t> *get_osd_stat() const {
+    return &parent.osd_stat;
+  }
+  const ceph::unordered_map<pg_t,pg_stat_t> *get_pg_stat() const {
+    return &parent.pg_stat;
+  }
+  float get_full_ratio() const { return parent.full_ratio; }
+  float get_nearfull_ratio() const { return parent.nearfull_ratio; }
+
+  bool have_creating_pgs() const { return !parent.creating_pgs.empty(); }
+  bool is_creating_pg(pg_t pgid) const { return parent.creating_pgs.count(pgid); }
+  virtual void maybe_add_creating_pgs(epoch_t scan_epoch,
+				      creating_pgs_t *pending_creates) const {
+    if (parent.last_pg_scan >= scan_epoch) {
+      for (auto& pgid : parent.creating_pgs) {
+      auto st = parent.pg_stat.find(pgid);
+      assert(st != parent.pg_stat.end());
+      auto created = make_pair(st->second.created, st->second.last_scrub_stamp);
+      // no need to add the pg, if it already exists in creating_pgs
+      pending_creates->pgs.emplace(pgid, created);
+      }
+    }
+  }
+  epoch_t get_min_last_epoch_clean() const { return parent.get_min_last_epoch_clean(); }
```

Comment:
Just making it explicit what functionality is actually used by the rest of the Monitor to help build up the interface. See the later commits where we ABC the PGStatService.